apps/prairielearn/python/zygote.py

```diff
@@ -84,6 +85,31 @@
 prairielearn.get_unit_registry()
 
 
+# We want to conditionally allow/block importing specific modules, namely `rpy2`.
+# This custom importer will allow us to do so, and throw a custom error message.
+#
+# While this won't prevent anything more complex than an `import` statement, it
+# will make it clear to the user that they're not allowed to use `rpy2`. If they
+# try to bypass the block, it's up to them to deal with the consequences.
+class ForbidModuleMetaPathFinder(importlib.abc.MetaPathFinder):
```

Comment:
Nit: Functions don't have any type annotations. Doesn't matter _too_ much since this is used one place but also an easy fix.

Reply:
Done in 942477a.